### DISCUSSION TITLE
Variadic macros issue with FW_ASSERT

### DISCUSSION THREAD

**Discussion by JackNWhite** [nasa/fprime#2409] [Q&A]
If you have built F' with the `-pedantic` option in GCC, you will have fallen foul of the following error, in many places:

```
warning: ISO C++11 requires at least one argument for the "..." in a variadic macro
   57 |     FW_ASSERT(this->header.type == T_END);
```

This is because the macro is defined thus: `#define FW_ASSERT(cond, ...)`

Standard C++ prior to C++ 20 requires that at least one thing be passed as the variable argument list (`...`), but most uses of the macro in F' don't do that.

Using the `-Wno-variadic-macros` hasn't worked for me (g++ 11.4.0). Has anyone been able to overcome this difficulty?

**Comment by Joshua-Anderson** [comment]
Correct, you cannot build F' with `-pedantic` on GCC. `-Wno-variadic-macros` only overrides `-pedantic` in clang. [F' CI checks](https://github.com/nasa/fprime/blob/v3.4.0/CMakeLists.txt#L43-L50) are setup to only use the pedantic flag when testing against clang.

I think it's worth revisiting how F' does assertions, but I'm not sure anyone has identified a better solution than variadic macros yet.
